src/tests/functional/plugin/shared/include/multi/multi_remote_blob_multidevice_test.hpp

```diff
@@ -80,3 +76,163 @@ TEST_P(MultiDeviceMultipleGPU_Test, canCreateRemoteTensorThenInferWithAffinity)
         }
     }
 }
+
+TEST_P(MultiDeviceCreateContextMultipleGPU_Test, canInferOnUserContextWithSystemMemory) {
+    auto ie = ov::Core();
+    ov::CompiledModel exec_net_regular;
+    try {
+        exec_net_regular = ie.compile_model(function, device_names);
+    } catch (...) {
+        // device is unavailable (e.g. for the "second GPU" test) or other (e.g. env) issues not related to the test
+        return;
+    }
+    auto input = function->get_parameters().at(0);
+    auto output = function->get_results().at(0);
+
+    // regular inference
+    auto inf_req_regular = exec_net_regular.create_infer_request();
+    auto fakeImageData = FuncTestUtils::create_and_fill_tensor(input->get_element_type(), input->get_shape());
+    inf_req_regular.set_tensor(input, fakeImageData);
+
+    inf_req_regular.infer();
+    auto output_tensor_regular = inf_req_regular.get_tensor(exec_net_regular.output());
+
+    // inference using remote tensor
+    std::vector<std::shared_ptr<OpenCL>> ocl_instances;
+    std::vector<cl::Device> devices;
+    for (int i = 0; i < device_lists.size(); i++) {
+        auto ocl_instance_tmp = std::make_shared<OpenCL>(i);
+        ocl_instances.push_back(ocl_instance_tmp);
+        devices.push_back(ocl_instances.back()->_device);
+    }
+    cl::Context multi_device_ctx(devices);
+    auto ocl_instance = std::make_shared<OpenCL>(multi_device_ctx.get());
+    std::vector<ov::RemoteContext> remote_contexts;
+    for (int i = 0; i < device_lists.size(); i++) {
+        auto remote_context = ov::intel_gpu::ocl::ClContext(ie, ocl_instance->_context.get(), i);
+        remote_contexts.push_back(remote_context);
+    }
+    ov::AnyMap context_list;
+    for (auto& iter : remote_contexts) {
+        context_list.insert({iter.get_device_name(), iter});
+    }
+    auto multi_context = ie.create_context(device_names, context_list);
```

Comment:
do I need to pass MULTI:GPU.0,GPU.1 here? I suppose context_list already has information that I have GPU.0 and GPU.1 here. 
So, I expect we can use `ie.create_context("MULTI", { remote_context_for_gpu_0, remote_context_for_gpu_1 })`, is not it? where each `remote_context_for_gpu_0` and `remote_context_for_gpu_1` are `ov::RemoteContext` and it means we want to compile mode for `MULTI: remote_context_for_gpu_0->get_device_name(), remote_context_for_gpu_1->get_device_name()`

Reply:
Hi Ilya, first, create_context only need MULTI/AUTO as the first param, here in the case, I just pass the variable device_names for convenience.
As your concern, it relates to 2 usage scenarios when trying to use remote context in multi/auto:
1) still use legacy key MULTI_DEVICE_PRIORITES for multi/auto plugin when deciding target hardwares, in this case, when loading from context, multi/auto will try to locate corresponding context from multi contexts, and if able to locate, then load to this hardware with context, else load without context
2) As you described above, when load to multi with context, the legacy key MULTI_DEVICE_PRIORITES can be ignored for multi/auto, we just load to devices which can be mapped in the multi remote context
From usage consistency perspective I would prefer option 1, and also, if use option 2, then we are not able to load to hardware that are not supporting remote context now, like CPU+GPU case

Reply:
Agree with Ilya. Also I think there's no need to make users call iter.get_device_name() to create map. So we need to come up with better syntax. 
Maybe one of the options below?
1. Use generic Core::create_context API: 
```
// remote_contexts property is needed to convert list to AnyMap
constexpr Property<...> ov::multi::remote_contexts;
auto multi_context = ie.create_context("MULTI", ov::multi::remote_contexts(ctx1, ctx2));
```
2. Introduce public remote context for multi:
```
// some public header
class MultiRemoteContext : public RemoteContext {
MultiRemoteContext(std::vector<RemoteContext> hw_contexts);
};
// app
ov::MultiRemoteContext multi_ctx({ctx1, ctx2});
core.compile_model(model, multi_ctx);
// It can also allow to allocate tensors for specific ctx:
auto tensor = multi_ctx.create_tensor(device_id_in_multi_context, element_type, shape);
```

Reply:
Hi @vladimir-paramuzov  thanks for the review! 
The get_device_name() is indeed an additional and unecesssary step for user to call in order to create multiremotecontext map.  I was trying to make the smallest change to implement this requirement by directly using the AnyMap param in create_context API,  but I agree to come up with a better syntax.

for option 1, maybe from property perspective it's not good to introduce a new property ov::multi::remote_contexts? @ilya-lavrenov what do you think?

I would prefer option 2, and, for create_tensor, I think user should create from separate hardware contexts directly, such as in GPU context, there are multiple APIs to create tensor, like from cl_mem/cl_buffer/usm_ptr/nv12, if we want to allocate tensors in multi for specific ctx id, we have to additionally wrap all kinds of tensor type information and pass to the target hardware context, do we really need to support this scenario?

Reply:
Usage scenarios:
1. Using default context to create memory, and load network without user context(e.g. default context is being used in gpu)
with recent remote context reuse pr in gpu plugin, the default context can be shared no matter whatever compile_model configs, and MULTI/AUTO can infer with created remote memory.

2. use memory allocated from default context and load network with user created context
     C++ exception with description "Check 'mem.is_allocated_by(net_engine)' failed at src/plugins/intel_gpu/src/graph/primitive_inst.cpp:105:
[GPU] Can't set memory due to engines mismatch. Network was created for 0x5579e60c12b0 (Intel(R) UHD Graphics 630 [0x3e92]) engine while memory object was allocated for 0x5579e5f9a420(Intel(R) UHD Graphics 630 [0x3e92])
" thrown in the test body.
@wangleis @vladimir-paramuzov  with second scenario, I suppose we will need to implement loadnetwork with ctx for multi.